docs/proposals/20250818-multi-az-apiserver-loadbalancer.md

```diff
@@ -0,0 +1,298 @@
+# Multi-AZ API Server LoadBalancer for CAPO
+
+## Summary
+Add first-class Multi-AZ support for the Kubernetes control plane LoadBalancer in Cluster API Provider OpenStack (CAPO). The feature reconciles one Octavia LoadBalancer per Availability Zone (AZ), places each VIP in the intended subnet for that AZ via an explicit AZ→Subnet mapping, and by default registers control plane nodes only with the LB in the same AZ. Operators expose the control plane endpoint via external DNS multi-value A records that point at the per-AZ LB IPs. This proposal is additive and backward compatible.
+
+## Motivation
+- Achieve true multi-AZ resilience for the control plane by avoiding a single VIP dependency.
+- Align control plane networking with existing multi-AZ compute placement goals.
+- Provide clear, portable primitives across Octavia providers with native AZ hints and an explicit, unambiguous mapping between AZs and VIP subnets.
+
+## Goals
+- Create and manage one API server LoadBalancer per configured AZ.
+- Support explicit AZ→Subnet mapping only (no positional mapping).
+- Default to same-AZ LB membership for control plane nodes; allow opt-in cross-AZ registration.
+- Keep the API additive with strong validation, clear events and documentation.
+- Preserve user-provided DNS endpoints; DNS record management remains out of scope.
+
+## Non-Goals
+- Managing or provisioning DNS records.
+- Provider-specific topologies such as ACTIVE_STANDBY across fault domains.
+- Service type LoadBalancer for worker Services.
```

Comment:
It seems to me that automatic creation of the network and subnets is also a non-goal, right?